examples/mediadevices/src/helpers.js

```diff
@@ -14,42 +15,54 @@ function getDevicesOfKind(deviceInfos, kind) {
   });
 }
 
+function switchLocalTracks(room, newLockTrack) {
+  if (room) {
+      room.localParticipant.tracks.forEach(function(trackPublication) {
+      if (trackPublication.kind === newLockTrack.kind) {
+        room.localParticipant.unpublishTrack(trackPublication.track);
+      }
+    });
+    room.localParticipant.publishTrack(newLockTrack);
+  }
+}
+
 /**
  * Apply the selected audio input device.
  * @param {string} deviceId
  * @param {HTMLAudioElement} audio
+ * @param {Room} [room] to switch tracks on.
  * @returns {Promise<void>}
  */
-function applyAudioInputDeviceSelection(deviceId, audio) {
+function applyAudioInputDeviceSelection(deviceId, audio, room) {
   return Video.createLocalAudioTrack({
-    deviceId: deviceId
+    deviceId: {
+      exact: deviceId // NOTE: on ios safari - it respects the deviceId only if its exact.
+    }
   }).then(function(localTrack) {
     localTrack.attach(audio);
+    switchLocalTracks(room, localTrack);
+  }).catch(function(error) {
+    console.log('applyAudioInputDeviceSelection failed:', error);
   });
 }
 
-/**
- * Apply the selected audio output device.
- * @param {string} deviceId
- * @param {HTMLAudioElement} audio
- */
-function applyAudioOutputDeviceSelection(deviceId, audio) {
```

Comment:
@makarandp0 I think you should retain this and only apply where `setSinkId` is supported.

Reply:
I moved this functionality out of the helper.js because a) this example is already getting  big, and more importantly the part of handling output devices does not require/demonstrates any of our SDK apis.

@manjeshbhargav - what do you think about removing output device selection from this sample. This is outside the scope of our API, and does not work on safari.
It also does not render waveform on mobile chrome either, I did not investigate why.

Reply:
@makarandp0 I think we should still retain it here because we have been asked question before as to how to redirect media to specific audio output devices. That is the reason we added this example in the first place. However, I would add an additional check for `setSinkId` support:
```js
/**
 * Apply the selected audio output device.
 * @param {string} deviceId
 * @param {HTMLAudioElement} audio
 * @returns {Promise<void>}
 */
function applyAudioOutputDeviceSelection(deviceId, audio) {
  return typeof audio.setSinkId === 'function'
    ? audio.setSinkId(deviceId)
    : Promise.reject('This browser does not support setting an audio output device');
}
```